### DISCUSSION TITLE
[Suggestion] Allow setting Wifi output power

### DISCUSSION THREAD

**Discussion by miguelruivo** [RavenSystem/esp-homekit-devices#2292] [New Features]
Do you think that it'll be possible to fine tune the wifi tx power in a future firmware? Asking because from a esp8266 perspective there is that parameter that can be configured up to 20 dbm (which is the default) and in a house full of relays, having all of them "scream" at 20 dbm can cause some interference that may be avoided. Having the default should be ok for most of the users, but having the possibility to fine tune it based on just a simple integer/drodpwn should allow for more fine tuning for those that want it. You can check it [here](https://gitlab.com/painlessMesh/painlessMesh/-/issues/51)  an example.

There should be a `Wifi.setOutputPower` method on esp8266 SDK that allows anything from 0 - 20.5, and defaults to 20.5 from what I understood.

**Comment by RavenSystem** [comment]
This doesn't make sense. Wifi devices don't transmit all at same time. Devices transmit one by one, because Wifi connection is a peer to peer connection from the radio physic perspective. When a device is sending data to the AP, all the others devices don't transmit anything. And other thing that I said sometime ago, TX power transmission is the maximum power that a device can use, and it is regulated automatically from the Wifi AP. Overriding it can be illegal. And FYI: There is not any way to override TX power in the open-rtos-sdk. But as I said, devices connected to the same Wifi can not create interferences between them.
